### PR TITLE
Moved DefaultHouseAd.GetAssemblyFromClass to ReflectionHelpers.GetAssemb...

### PR DESCRIPTION
**Pull request by rafalkozik**
...lyFromClassName, improved implementation and added unit tests.

Tests/TestData and TestData.Nested class libraries added for testing purposes. AdRotator2013.sln needs to be updated, unfortunately I don't have access to Visual Studio 2013.

### CODE REVIEW

File: AdRotator/DefaultHouseAd.cs [hunk 1]
```diff
@@ -58,7 +58,9 @@ public DefaultHouseAd(int AdRotatorControlID, IFileHelpers FileHelper)
             Object o = null;
             try
             {
-                var asm = GetAssemblyFromClass(LocalHouseAdBodyName);
+                ReflectionHelpers reflectionHelpers = new ReflectionHelpers();
```
Comment (SimonDarksideJ) [inline AdRotator/DefaultHouseAd.cs]: Hmm, why new up the Reflection helpers?
The majority of the reflectionhelpers are static functions in a static class, no need to instantiate.

Can this be updated?

File: AdRotator/Utilities/ReflectionHelpers.cs [hunk 2]
```diff
@@ -267,6 +267,35 @@ private Type GetDelegateReturnType(Type d)
 
             return invoke.ReturnType;
         }
+
+        public Assembly GetAssemblyFromClassName(string className)
```
Comment (SimonDarksideJ) [inline AdRotator/Utilities/ReflectionHelpers.cs]: This needs to be a static class

Comment (rafalkozik) [inline AdRotator/Utilities/ReflectionHelpers.cs]: I made GetAssemblyFromClassName static, making ReflectionHelpers a static
class is a separate task.

**Comment by SimonDarksideJ** [comment]
Thanks @rafalkozik I'll get this checked tonight and report back.
Looks good so far.

Have you confirmed through local testing and UnitTests that the Class name resolution for house ads is now working as expected?

**Comment by rafalkozik** [comment]
Yes, name resolution is verified through UnitTests, 3 of 4 of tested calls
would provide invalid result with the previous implementation.

**Pull request merged**